packages/kit/src/utils/rpc.ts

```diff
@@ -20,18 +23,39 @@ export async function getRpcHandler<
   definition: RpcFunctionDefinition<NAME, TYPE, ARGS, RETURN>,
   context: DevToolsNodeContext,
 ): Promise<(...args: ARGS) => RETURN> {
+  let handler: (...args: ARGS) => RETURN
+
   if (definition.handler) {
-    return definition.handler
+    handler = definition.handler
   }
-  if (definition.__resolved?.handler) {
-    return definition.__resolved.handler
+  else if (definition.__resolved?.handler) {
+    handler = definition.__resolved.handler
   }
-  definition.__promise ??= Promise.resolve(definition.setup(context))
-    .then((r) => {
-      definition.__resolved = r
-      definition.__promise = undefined
-      return r
-    })
-  const result = definition.__resolved ??= await definition.__promise
-  return result.handler
+  else {
+    definition.__promise ??= Promise.resolve(definition.setup(context))
+      .then((r) => {
+        definition.__resolved = r
+        definition.__promise = undefined
+        return r
+      })
+    const result = definition.__resolved ??= await definition.__promise
+    handler = result.handler
+  }
+
+  return ((...args: ARGS) => {
+    try {
+      const result = handler(...args)
+      if (result instanceof Promise) {
+        return result.catch((error) => {
+          debug(`RPC call "${definition.name}" failed:`, error)
```

Comment:
Error should not go to debug (which is default to silent). We could have a logger interface where for now could be a simple abstraction of `console.log` and `console.error`